lib/src/pretty_qr_data_view.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:qr/qr.dart';
+import 'package:meta/meta.dart';
+import 'package:flutter/widgets.dart';
+
+import 'package:pretty_qr_code/src/pretty_qr_view.dart';
+import 'package:pretty_qr_code/src/painting/pretty_qr_decoration.dart';
+
+/// {@macro pretty_qr_code.PrettyQrView}
+@internal
+class PrettyQrDataView extends StatefulWidget {
+  /// The QR code data.
+  @protected
+  final String data;
```

Comment:
We use string for data cache?

Reply:
Yep, that should be enough for most cases. Otherwise, you can use the `PrettyQrView` constructor directly.